src/sources/host_metrics/filesystem.rs

```diff
@@ -7,44 +7,31 @@ use vector_core::metric_tags;
 
 use crate::internal_events::{HostMetricsScrapeDetailError, HostMetricsScrapeFilesystemError};
 
-use super::{filter_result, FilterList, HostMetrics};
+use super::{default_all_devices, example_devices, filter_result, FilterList, HostMetrics};
 
 /// Options for the “filesystem” metrics collector.
 #[configurable_component]
 #[derive(Clone, Debug, Default)]
 pub struct FilesystemConfig {
     /// Lists of device name patterns to include or exclude in gathering
     /// usage metrics.
-    ///
-    /// Defaults to including all devices.
-    #[serde(default)]
```

Comment:
Since this (and the others) changed to an explicit serde default, does the logic that is handling these config options need to be updated to not check if it's empty to apply a default there?

Reply:
cc: @spencergilbert since your `hidden` attribute comment is about a section of code with a similar problem.

So this is a bit of a tricky situation. The old docs specified `*` as the default, but that value was not actually used as the default in the code, an empty `FilterList` was. With these changes I made, the default will now be a `FilterList` that includes strings matching `*`. That's theoretically still valid behavior, and afaik the code handles `*` correctly in all cases, but it is changing the default behavior slightly (new globbing codepaths will now always execute).

If we want to change up the behavior, I could move `*` into the examples section. With serde, I coudn't find a simple way to specific a default, but not use it during deserialization (seems like a bad pattern anyway).

Reply:
> new globbing codepaths will now always execute

What exactly do you mean by this?

Reply:
Both an empty FilterList and a `*` FilterList lead to the same behavior. The empty FilterList leads to "all" style behavior because we don't execute any filters, then a `*` FilterList includes all devices or paths since this globbing logic runs: https://github.com/vectordotdev/vector/blob/dh%2Fhost_metrics_autogen_docs/src/sources/host_metrics/mod.rs#L495

All I'm trying to say is that the behavior is unchanged. Previously, we relied on the empty FilterList behavior since the docs were separate from the code. Now that the code is the source of truth, `*` is now the default, but the behavior is the same.

Reply:
Ah, yeah gotcha - I think it sounds find as you've changed it then?

Reply:
Yeah, I think it's fine to leave the new default. Any thoughts, @neuronull?

Reply:
Well, honestly it almost feels like the `includes` are not actually `Optional` - as you pointed out, we just use `*` if nothing is provided. The component _needs_ some includes pattern, the user just doesn't have to provide it. Perhaps the `*` should be the `serde(default)`?

Reply:
I think I follow. Since the behavior is unchanged and the docs were instead upgraded, I wonder if this deserves a mention in the upgrade guide? Or if we don't usually do that unless behavior changed. A user without our knowledge might see the docs changing and think the behavior should change, perhaps calling out that the docs were incorrect would be helpful.

Reply:
I don't think it's worth the note personally.